thorlcr/graph/thgraphslave.cpp

```diff
@@ -906,15 +906,16 @@ bool CSlaveGraph::recvActivityInitData(size32_t parentExtractSz, const byte *par
             actInitRtnData.clear();
             actInitRtnData.append(true);
             serializeThorException(e, actInitRtnData);
-            e->Release();
-            ret = false;
+            exception.setown(e);
         }
     }
     if (syncInitData() || needActInit)
     {
         if (!queryJobChannel().queryJobComm().send(actInitRtnData, 0, replyTag, LONGTIMEOUT))
```

Comment:
would perhaps the LONGTIMEOUT be less if the exception was set ?

Reply:
not really related, if this send fails it's likely the master is not responding and all best are off.